src/test-workspace.ts

```diff
@@ -0,0 +1,162 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import chalk from 'chalk';
+import * as path from 'path';
+import {run, WorkspaceRepo} from 'polymer-workspaces';
+
+import {WorkspaceConversionSettings} from './convert-workspace';
+import {generatePackageJson, localDependenciesBranch, readJson, writeJson} from './manifest-converter';
+import {lookupNpmPackageName} from './urls/workspace-url-handler';
+import {exec, logRepoError, logStep} from './util';
+
+/**
+ * Configuration options required for workspace testing. Same as conversion
+ * settings.
+ */
+interface WorkspaceTestSettings extends WorkspaceConversionSettings {}
+
+/**
+ * Setup the workspace repos for testing. Be sure to call restoreRepos() after
+ * testing is complete.
+ */
+async function setupRepos(
+    reposUnderTest: WorkspaceRepo[],
+    localConversionMap: Map<string, string>,
+    options: WorkspaceTestSettings) {
+  // Yarn sometimes has some issues with bad packages entering the cache from
+  // modulizer. Clear the cache before starting.
+  const {stderr} = await exec(options.workspaceDir, 'yarn', ['cache', 'clean']);
+  if (stderr.length > 0) {
+    console.log(chalk.dim(`yarn cache clean: ${stderr}`));
+  }
+
+  const batchResults = await run(reposUnderTest, async (repo) => {
+    await exec(repo.dir, 'git', ['checkout', '-B', localDependenciesBranch]);
+    writeTestingPackageJson(repo, localConversionMap, options.packageVersion);
+    await exec(
+        repo.dir,
+        'git',
+        ['commit', '-a', '-m', 'testing commit', '--allow-empty']);
+  }, {concurrency: 10});
+  batchResults.failures.forEach(logRepoError);
+
+  return [...batchResults.successes.keys()];
```

Comment:
Why not return the whole result and let the caller ignore failures if they want to. Otherwise there's no way for the caller to know if there were failures.